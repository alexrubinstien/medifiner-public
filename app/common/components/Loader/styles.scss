@import './app/common/styles/utils/_vars.scss';

.loader {
  border: 4px solid $border-color;
  border-top: 4px solid $primary-color;
  border-radius: 50%;
  width: 25px;
  height: 25px;
  animation: spin 2s linear infinite;

  &__push_left {
    margin-left: 96px;
  }
}
@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}