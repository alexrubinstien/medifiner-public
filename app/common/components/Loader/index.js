import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.scss';

const Loader = ({ className, ...props }) => (
    <div
      className={classNames(className, {
        loader: true,
      })}
    >
    </div>
  );

Loader.propTypes = {
  className: PropTypes.string,
};

Loader.defaultProps = {
  className: '',
};

export default Loader;
