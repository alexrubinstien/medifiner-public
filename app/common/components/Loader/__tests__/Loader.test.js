/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../index.js';

describe('With Snapshot Testing', () => {
  it('Loader component', () => {
    const component = renderer.create(
      <Loader>
        Loader
      </Loader>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Loader component with custom class', () => {
    const component = renderer.create(
      <Loader className="test-class">
        Loader
      </Loader>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
