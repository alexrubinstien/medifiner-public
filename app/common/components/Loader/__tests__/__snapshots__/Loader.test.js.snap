// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Loader component 1`] = `
<loader>
  Loader
</loader>
`;

exports[`With Snapshot Testing Loader component with custom class 1`] = `
<loader
  className="test-class loader"
>
  Loader
</loader>
`;
